File: video-series-query/src/main/java/org/jesperancinha/video/query/handlers/VideoSeriesEventHandler.java
package org.jesperancinha.video.query.handlers;

import org.axonframework.config.ProcessingGroup;
import org.axonframework.eventhandling.EventHandler;
import org.axonframework.queryhandling.QueryHandler;
import org.jesperancinha.video.core.data.VideoSeriesDto;
import org.jesperancinha.video.core.events.AddSeriesEvent;
import org.jesperancinha.video.query.queries.FindAllVideoSeriesQuery;
import org.jesperancinha.video.query.data.VideoSeries;
import org.jesperancinha.video.query.jpa.VideoSeriesRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.stream.Collectors;

@Service
@ProcessingGroup("video-series")
public class VideoSeriesEventHandler {

    private final VideoSeriesRepository videoSeriesRepository;

    public VideoSeriesEventHandler(VideoSeriesRepository videoSeriesRepository) {
        this.videoSeriesRepository = videoSeriesRepository;
    }

    @EventHandler
    @SuppressWarnings("unused")
    public void on(AddSeriesEvent event) {
        videoSeriesRepository.save(VideoSeries
                .builder()
                .name(event.getName())
                .volumes(event.getVolumes())
                .genre(event.getGenre())
                .cashValue(event.getCashValue())
                .build());
    }

    @QueryHandler
    @SuppressWarnings("unused")
    public List<VideoSeriesDto> handle(FindAllVideoSeriesQuery query) {
        return videoSeriesRepository.findAll().stream().map(
                videoSeries -> VideoSeriesDto.builder()
                        .name(videoSeries.getName())
                        .volumes(videoSeries.getVolumes())
                        .cashValue(videoSeries.getCashValue())
                        .genre(videoSeries.getGenre())
                        .build()).collect(Collectors.toList());
    }

}

File: video-series-query/src/main/kotlin/org/jesperancinha/video/query/handlers/VideoSeriesEventHandler.kt
package org.jesperancinha.video.query.handlers

import org.axonframework.config.ProcessingGroup
import org.axonframework.eventhandling.EventHandler
import org.axonframework.queryhandling.QueryHandler
import org.jesperancinha.video.core.data.VideoSeriesDto
import org.jesperancinha.video.core.events.AddSeriesEvent
import org.jesperancinha.video.query.data.VideoSeries
import org.jesperancinha.video.query.jpa.VideoSeriesRepository
import org.jesperancinha.video.query.queries.FindAllVideoSeriesQuery
import org.springframework.stereotype.Service
import java.util.stream.Collectors

@Service
@ProcessingGroup("video-series")
class VideoSeriesEventHandler(private val videoSeriesRepository: VideoSeriesRepository) {
    @EventHandler
    @Suppress("unused")
    fun on(event: AddSeriesEvent) {
        videoSeriesRepository.save(
            VideoSeries(
                name = event.name,
                volumes = event.volumes,
                genre = event.genre,
                cashValue = event.cashValue
            )
        )
    }

    @QueryHandler
    @Suppress("unused")
    fun handle(query: FindAllVideoSeriesQuery?): List<VideoSeriesDto> {
        return videoSeriesRepository.findAll().filterNotNull().map { videoSeries ->
            VideoSeriesDto(
                name = videoSeries.name,
                volumes = videoSeries.volumes,
                genre = videoSeries.genre,
                cashValue = videoSeries.cashValue
            )
        }
    }
}